.pfnci/schema.yaml

```diff
@@ -0,0 +1,342 @@
+# Python
+###############################################################################
+# Python version to test with.
+# `spec` must be a version specifier accepted by pyenv.
+# When adding a new Python version, you may also need to add the version to the
+# list of supported Python versions in the Python libraries.
+python:
+  "3.6":
+    spec: "3.6.14"
+  "3.7":
+    spec: "3.7.11"
+  "3.8":
+    spec: "3.8.11"
+  "3.9":
+    spec: "3.9.6"
+  "pre":
+    spec: "3.10.0b3"  # note that dependency python libraies may not work
+
+###############################################################################
+# Python Libraries
+###############################################################################
+# Library versions to test with.
+# `spec` must be a version specifier accepted by pip.
+# When adding a new NumPy version, you may also need to add the version to the
+# list of supported Python versions in SciPy.
+numpy:
+  "1.17":
+    spec: "==1.17.*"
+    python:
+      # - "3.5"
+      - "3.6"
+      - "3.7"
+      - "3.8"
+  "1.18":
+    spec: "==1.18.*"
+    python:
+      # - "3.5"
+      - "3.6"
+      - "3.7"
+      - "3.8"
+  "1.19":
+    spec: "==1.19.*"
+    python:
+      - "3.6"
+      - "3.7"
+      - "3.8"
+      - "3.9"
+  "1.20":
+    spec: "==1.20.*"
+    python:
+      - "3.7"
+      - "3.8"
+      - "3.9"
+  "1.21":
+    spec: "==1.21.*"
+    python:
+      - "3.7"
+      - "3.8"
+      - "3.9"
+  "pre":
+    spec: ">=0a0"  # latest, including pre-releases
+scipy:
+  null:
+    spec: null
+  "1.4":
+    spec: "==1.4.*"
+    python:
+      # - "3.5"
+      - "3.6"
+      - "3.7"
+      - "3.8"
+    numpy:
+      # - "1.13"
+      # - "1.14"
+      # - "1.15"
+      # - "1.16"
+      - "1.17"
+      - "1.18"
+      - "1.19"
+      - "1.20"
+      - "1.21"
+  "1.5":
+    spec: "==1.5.*"
+    python:
+      - "3.6"
+      - "3.7"
+      - "3.8"
+      - "3.9"
+    numpy:
+      # - "1.16"
+      - "1.17"
+      - "1.18"
+      - "1.19"
+      - "1.20"
+      - "1.21"
+  "1.6":
+    spec: "==1.6.*"
+    python:
+      - "3.7"
+      - "3.8"
+      - "3.9"
+    numpy:
+      # - "1.16"
+      - "1.17"
+      - "1.18"
+      - "1.19"
+      - "1.20"
+      - "1.21"
+  "1.7":
+    spec: "==1.7.*"
+    python:
+      - "3.7"
+      - "3.8"
+      - "3.9"
+    numpy:
+      # - "1.16"
+      - "1.17"
+      - "1.18"
+      - "1.19"
+      - "1.20"
+      - "1.21"
+  "pre":
+    spec: ">=0a0"  # latest, including pre-releases
+optuna:
+  null:
+    spec: null
+  "2":
+    spec: "==2.*"
+  "pre":
+    spec: ">=0a0"  # latest, including pre-releases
+cython:
+  "0.29":
+    spec: "==0.29.*"
+  "pre":
+    spec: ">=0a0"  # latest, including pre-releases
+
+###############################################################################
+# Accelerators
+###############################################################################
+# Corresponds to the environment variable "CUPY_ACCELERATORS".
```

Comment:
How about directly supporting other environment variables instead?, I think the needed change is negligible and the benefits are great.

Reply:
The current design is to define a list of **all possible values** in schema so that the coverage can be illustrated in [2-d matrix](https://github.com/kmaehashi/cupy/blob/add-ci-generator/.pfnci/coverage.md).
But I understand your point that adding a new env var test axis should be easy.
How about automatically handling all `env:*` in schema keys as environment variable axis? e.g.,

```yaml
env:CUPY_ACCELERATORS:
  null:
  "cub":
  "cutensor":
  "cub,cutensor":
  "cutensor,cub":
env:CUPY_USE_CUDA_PYTHON:
  null:
  "1":
```

Reply:
this looks cool, thanks!